packages/react-components/src/components/Picker/Picker.tsx

```diff
@@ -180,29 +180,27 @@ export const Picker: React.FC<IPickerProps> = ({
           isRequired={isRequired}
           isMultiSelect={type === 'multi'}
           size={size}
-          onTrigger={handleOnTrigger}
-          onClear={handleOnClear}
-          onFilter={handleOnFilter}
+          onTrigger={handleTrigger}
+          onClear={handleClear}
         >
-          {selected ? (
-            <TriggerBody
-              items={selected}
-              type={type}
-              onItemRemove={handleItemRemove}
-            />
-          ) : (
-            placeholder
-          )}
+          <TriggerBody
+            isOpen={isListOpen}
+            isSearchDisabled={searchDisabled}
+            placeholder={placeholder}
+            items={selected}
+            type={type}
+            onItemRemove={handleItemRemove}
+            onFilter={handleOnFilter}
+          />
         </Trigger>
         <PickerList
           selectedItemsKeys={selected ? getSelectedItemsKeys(selected) : null}
```

Comment:
I think we can move this whole condition inside a memoized array `selectedItemKeys`. You can use `useMemo` here.